eva_submission/eload_ingestion.py

```diff
@@ -16,33 +37,71 @@ class EloadIngestion(Eload):
 
     def __init__(self, eload_number):
         super().__init__(eload_number)
-        self.eload_cfg.set(self.config_section, 'ingestion_date', value=self.now)
         self.settings_xml_file = cfg['maven_settings_file']
+        self.project_accession = self.eload_cfg.query('brokering', 'ena', 'PROJECT')
+        self.project_dir = self.setup_project_dir()
+        self.pg_uri = get_pg_metadata_uri_for_eva_profile('development', self.settings_xml_file)
+        self.mongo_uri = get_mongo_uri_for_eva_profile('production', self.settings_xml_file)
 
-    def ingest(self, db_name=None, tasks=None):
+    def ingest(
+            self,
+            aggregation=None,
+            instance_id=None,
+            vep_version=None,
+            vep_cache_version=None,
+            db_name=None,
+            tasks=None
+    ):
         # TODO assembly/taxonomy insertion script should be incorporated here
+        # TODO set ENA data release date
+        self.eload_cfg.set(self.config_section, 'ingestion_date', value=self.now)
         self.check_variant_db(db_name)
 
         if not tasks:
             tasks = self.all_tasks
 
         if 'metadata_load' in tasks:
             self.load_from_ena()
-        if 'accession' in tasks:
-            self.warning('Accessioning not yet supported, skipping.')
-        if 'variant_load' in tasks:
-            self.warning('Variant loading not yet supported, skipping.')
+        do_accession = 'accession' in tasks
+        do_variant_load = 'variant_load' in tasks
+
+        if do_accession or do_variant_load:
+            aggregation = aggregation.lower()
+            if aggregation not in {'basic', 'none'}:
+                raise ValueError('Aggregation type must be BASIC or NONE')
```

Comment:
Is this required since it's already tested in the CLI?

Reply:
indeed, removed